src/utilities.ts

```diff
@@ -30,3 +30,67 @@ export async function getDefaultExport(
   }
   return functionModule.default;
 }
+
+/**
+ * Performs a specified action, encapsulated in a provided callback function, for each function
+ * defined in the application manifest. Performs basic validation on each function definition;
+ * if any definition fails validation, an exception will be thrown.
+ * @param {string} applicationRoot - Absolute path to application root directory.
+ * @param {any} manifest - An object representing the application manifest. Should contain a `functions` key that is a map of function IDs to function definitions.
+ * @param {(functionID: string, functionDefinition: any) => Promise<void>} callbackFn - A function to be invoked once per defined function present in the application manifest. Arguments provided to this function are: a function ID string, a function definition object, and a validated path to the function's source file.
+ */
+export async function forEachValidatedManifestFunction(
```

Comment:
Let me know if this utility name is weird. I spent like 15 minutes trying to come up with a good one and this was the best I could come up with - which is not great!